File: src/main/java/com/network/model/NetworkDevice.java
package com.network.model;

import lombok.Data;
import javax.persistence.*;

@Data
@Entity
public class NetworkDevice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String macAddress;
    private String ipAddress;
    private String deviceName;
    private String deviceType;
    private boolean isOnline;
    
    @OneToOne(cascade = CascadeType.ALL)
    private NetworkMetrics metrics;
} 

File: src/frontend/src/types/NetworkDevice.ts
export interface NetworkDevice {
    id: number;
    macAddress: string;
    ipAddress: string;
    deviceName: string;
    deviceType: string;
    isOnline: boolean;
    metrics?: NetworkMetrics;
}

export interface NetworkMetrics {
    bandwidth: number;
    latency: number;
    packetLoss: number;
    timestamp: string;
}

export interface NetworkConfiguration {
    ip: string;
    mask: string;
    gateway?: string;
    dns?: string[];
} 